connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -258,6 +255,7 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
         allGroups.addAll(configDef.groups());
         configValues.addAll(config.configValues());
 
+        Plugins.compareAndSwapLoaders(save);
```

Comment:
Should this be called in a `finally` block? Right now, if there are any exceptions after `ClassLoader save = worker.getPlugins().compareAndSwapLoaders(connector);` but before line 258, then line 258 will not be set. Is that a problem?